### ISSUE TITLE
Make more information available about the hardware configuration (maybe from lshw and hwloc-ls))

### ISSUE THREAD

**Issue by wcohen** [performancecopilot/pcp#286]
Currently PCP provides some information about hardware configuration through hinv.*.* such as number of processors (hinv.ncpu) and number of disk (hinv.ndisk).  There is also some information about the amount of memory on  each NUMA node (mem.numa.util.total).  However, it would be useful to have more details about the machine.  For example what particular model of disk drives are in the machine to better estimate the number of IOPS/s that a drive should be capable of.  The max IOP/s values differ significantly between 7.2K, 10K, 15K, and SSD drives. 

It can also be useful to know the topology of the system to identify bottlenecks due to high speed devices sharing the same bus, a task assigned to one numa node using a network device on another numa node, or multiple cpus on a single socket saturating the memory bus.

lshw command provides detailed model information, but it needs to be run privileged to get the model information. hwloc-ls provide a text output of the machine layout (lstopo provides a graphic version).

Would it be make the information provided by lshw and hwloc-ls into a pmda for pcp?

**Comment by natoscott** [comment]
| Would it be [possible to?] make the information provided by lshw and hwloc-ls into a pmda for pcp?

Yep.  Looks like both commands have XML output options, so could use an approach similar to the gluster PMDA (src/pmdas/gluster/pmdagluster.python).

**Comment by kmcdonell** [comment]
On my mickey mouse workstation, lshw produces 1500+ lines of output ... lots of these seem to have nothing to do with performance that I can see, so I think some culling of the data (perhaps based on a default opt-out policy) may be required until someone articlates as plausible use case for additional classes of information.

**Comment by natoscott** [comment]
@kmcdonell yeah - using the XML output, its relatively straightforward to pick out just the specific values of interest - pmdagluster does this kinda thing using the xmltree module, which can find XML tree elements using named paths.
